src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1546,7 +1550,7 @@ def create_vm(vm_name, resource_group_name, image=None, size='Standard_DS1_v2',
     vm_dependencies = []
     if storage_account_type == 'new':
         storage_account = storage_account or 'vhdstorage{}'.format(
-            random_string(14, force_lower=True))
+            hash_string(vm_id, length=14, force_lower=True))
```

Comment:
That does raise the issue in #3589.  Finding and reusing the storage account 1) isn't guaranteed (maybe it finds a different storage account) and 2) is arguable whether it is idempotent since running the same command twice results in different ARM templates.

Here yes I basically did it for consistency.